### ISSUE TITLE
Undefined constant Illuminate\Http\Request::HEADER_X_FORWARDED_ALL

### ISSUE THREAD

**Issue by qdmitry** [laravel/framework#40883]
- Laravel Version: 9.0
- PHP Version: 8.1.2
- Database Driver & Version: mysql 8.0

### Description: 
I install latest version `composer require fideloper/proxy:^4.4.1`, but error: 
_Undefined constant Illuminate\Http\Request::HEADER_X_FORWARDED_ALL_

I don't understand why the code doesn't match the master branch.

### Steps To Reproduce:

Screenshots

<img width="1239" alt="Снимок экрана 2022-02-09 в 11 48 31" src="https://user-images.githubusercontent.com/28624280/153141042-470731c4-3682-4174-b435-2fe52c388ae9.png">

<img width="1659" alt="Снимок экрана 2022-02-09 в 12 06 59" src="https://user-images.githubusercontent.com/28624280/153141075-1603428c-e4f2-415e-88b5-145c43887fd7.png">

composer.json
```
"require": {
        "php": "^8.0.2",
        "doctrine/dbal": "^3.0",
        "fideloper/proxy": "^4.4.1",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.4.1",
        "intervention/image": "^2.5",
        "laravel/framework": "^9.0",
        "laravel/sanctum": "^2.14",
        "laravel/tinker": "^2.7",
        "spatie/laravel-permission": "^5.0"
    },
    "require-dev": {
        "spatie/laravel-ignition": "^1.0",
        "fakerphp/faker": "^1.9.1",
        "laravel/breeze": "^1.8",
        "laravel/sail": "^1.13.2",
        "mockery/mockery": "^1.5.0",
        "nunomaduro/collision": "^6.0",
        "phpunit/phpunit": "^9.5.0"
    },
```

**Comment by pragnesh87** [comment]
https://laravel.com/docs/9.x/upgrade#upgrade-9.0
In document *Trusted Proxies* update steps already given. 
Please try those steps.

![image](https://user-images.githubusercontent.com/1000852/153142704-7cae4a5f-e88e-4735-ab1a-3790fa5cf9fa.png)

**Comment by driesvints** [comment]
Thanks @pragnesh87

**Comment by qdmitry** [comment]
Take a closer look at the problem! My project files are fine. The problem is loading from composer.

**Comment by qdmitry** [comment]
@driesvints check

<img width="1664" alt="Снимок экрана 2022-02-09 в 12 38 29" src="https://user-images.githubusercontent.com/28624280/153143687-87bc16c3-b973-4f96-9226-59d837174994.png">

**Comment by qdmitry** [comment]
@driesvints Need help! Stop ignoring it! This is a magical mistake, I can't handle it! Open the bug.

**Comment by pragnesh87** [comment]
Please make following changes
file: app/Http/Middleware/TrustProxies.php

```
use Fideloper\Proxy\TrustProxies as Middleware 
to 
use Illuminate\Http\Middleware\TrustProxies as Middleware.
```

**Comment by qdmitry** [comment]
@pragnesh87 It turns out that the `fideloper/proxy` package is no longer required.
I change use import.

**Comment by driesvints** [comment]
@qdmitry please read the upgrade guide carefully before posting issues.

**Comment by almaz-free** [comment]
you need to delete file "config/trustedproxy.php"

**Comment by spaceemotion** [comment]
@driesvints the upgrade guide does not explicitly state that the package is no longer required and can be deleted. I ran into the same issue yesterday. 

Even after changing out the middleware, the fideloper-package still auto-loaded its own config and ran into the error.

Maybe adding a text about removing the package would help?

**Comment by driesvints** [comment]
@spaceemotion I see. I was a bit confused because the screenshot of the OP still showed the incorrect namespace even though it was listed in the upgrade guide. I'll PR that to indicate you need to remove the package.

**Comment by HadiNiazi** [comment]
If the above answer does not work for you and you are getting the same error, execute one more line for removing the trusted/proxy from composer.json.
~~~
// Before...
protected $headers = Request::HEADER_X_FORWARDED_ALL;
 
// After...
protected $headers =
    Request::HEADER_X_FORWARDED_FOR |
    Request::HEADER_X_FORWARDED_HOST |
    Request::HEADER_X_FORWARDED_PORT |
    Request::HEADER_X_FORWARDED_PROTO |
    Request::HEADER_X_FORWARDED_AWS_ELB;
~~~
This will remove the proxy from the composer.json and the exception will be gone.
~~~
composer remove trusted/proxy
~~~
it's properly mentioned in the laravel 9 about fideloper/proxy (same way you can remove trusted proxy) in upgrade guide. https://laravel.com/docs/9.x/upgrade#the-assert-deleted-method

**Comment by Irfan-Majid** [comment]
when in laravel 10 i went to use it it shows this error 


  Problem 1
    - Root composer.json requires illuminate/contracts 6.0, found illuminate/contracts[v6.0.0] but these were not loaded, likely because it conflicts with another require.
  Problem 2
    - fideloper/proxy 1.0.0 requires illuminate/support 4.0.x -> found illuminate/support[v4.0.0, ..., v4.0.10] but these were not loaded, likely because it conflicts with another require.
    - fideloper/proxy[2.0.1, ..., 2.0.2] require illuminate/support ~4 -> found illuminate/support[v4.0.0, ..., v4.2.17] but these were not loaded, likely because it conflicts with another require.
    - fideloper/proxy[2.0.3, ..., 2.1.1] require illuminate/support ~4.0|~5.0 -> found illuminate/support[v4.0.0, ..., v4.2.17, v5.0.0, ..., v5.8.36] but these were not loaded, likely because it conflicts with another require.
    - fideloper/proxy[3.0.0, ..., 3.3.4, 4.0.0, ..., 4.1.0] require illuminate/contracts ~5.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36] but it conflicts with your root composer.json require (6.0).
    - fideloper/proxy 4.2.0 requires illuminate/contracts ~5.0|~6.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44] but these were not loaded, likely because it conflicts with another require.
    - fideloper/proxy[4.2.1, ..., 4.2.2] require illuminate/contracts ^5.0|^6.0|^7.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6] but these were not loaded, likely because it conflicts with another require.
    - fideloper/proxy[4.3.0, ..., 4.4.0] require illuminate/contracts ^5.0|^6.0|^7.0|^8.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but these were not loaded, likely because it conflicts with another require.
    - fideloper/proxy[4.4.1, ..., 4.4.2] require illuminate/contracts ^5.0|^6.0|^7.0|^8.0|^9.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.7] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires fideloper/proxy * -> satisfiable by fideloper/proxy[1.0.0, 2.0.1, ..., 2.1.1, 3.0.0, ..., 3.3.4, 4.0.0, ..., 4.4.2].

**Comment by tisuchi** [comment]
@all Those who are facing this issue, make sure that you have changed the `config/trustedproxy.php` file also. 

```diff
<?php

return [

    /*
     * Set trusted proxy IP addresses.
     *
     * Both IPv4 and IPv6 addresses are
     * supported, along with CIDR notation.
     *
     * The "*" character is syntactic sugar
     * within TrustedProxy to trust any proxy
     * that connects directly to your server,
     * a requirement when you cannot know the address
     * of your proxy (e.g. if using ELB or similar).
     *
     */
    'proxies' => null, // [<ip addresses>,], '*', '<ip addresses>,'

    /*
     * To trust one or more specific proxies that connect
     * directly to your server, use an array or a string separated by comma of IP addresses:
     */
    // 'proxies' => ['192.168.1.1'],
    // 'proxies' => '192.168.1.1, 192.168.1.2',

    /*
     * Or, to trust all proxies that connect
     * directly to your server, use a "*"
     */
    // 'proxies' => '*',

    /*
     * Which headers to use to detect proxy related data (For, Host, Proto, Port)
     *
     * Options include:
     *
     * - Illuminate\Http\Request::HEADER_X_FORWARDED_ALL (use all x-forwarded-* headers to establish trust)
     * - Illuminate\Http\Request::HEADER_FORWARDED (use the FORWARDED header to establish trust)
     * - Illuminate\Http\Request::HEADER_X_FORWARDED_AWS_ELB (If you are using AWS Elastic Load Balancer)
     *
     * - 'HEADER_X_FORWARDED_ALL' (use all x-forwarded-* headers to establish trust)
     * - 'HEADER_FORWARDED' (use the FORWARDED header to establish trust)
     * - 'HEADER_X_FORWARDED_AWS_ELB' (If you are using AWS Elastic Load Balancer)
     *
     * @link https://symfony.com/doc/current/deployment/proxies.html
     */
-     'headers' => Illuminate\Http\Request::HEADER_X_FORWARDED_ALL,

];


```

**Comment by summerKK** [comment]
> when in laravel 10 i went to use it it shows this error
> 
> Problem 1 - Root composer.json requires illuminate/contracts 6.0, found illuminate/contracts[v6.0.0] but these were not loaded, likely because it conflicts with another require. Problem 2 - fideloper/proxy 1.0.0 requires illuminate/support 4.0.x -> found illuminate/support[v4.0.0, ..., v4.0.10] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[2.0.1, ..., 2.0.2] require illuminate/support ~4 -> found illuminate/support[v4.0.0, ..., v4.2.17] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[2.0.3, ..., 2.1.1] require illuminate/support ~4.0|~5.0 -> found illuminate/support[v4.0.0, ..., v4.2.17, v5.0.0, ..., v5.8.36] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[3.0.0, ..., 3.3.4, 4.0.0, ..., 4.1.0] require illuminate/contracts ~5.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36] but it conflicts with your root composer.json require (6.0). - fideloper/proxy 4.2.0 requires illuminate/contracts ~5.0|~6.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[4.2.1, ..., 4.2.2] require illuminate/contracts ^5.0|^6.0|^7.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[4.3.0, ..., 4.4.0] require illuminate/contracts ^5.0|^6.0|^7.0|^8.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[4.4.1, ..., 4.4.2] require illuminate/contracts ^5.0|^6.0|^7.0|^8.0|^9.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.7] but these were not loaded, likely because it conflicts with another require. - Root composer.json requires fideloper/proxy * -> satisfiable by fideloper/proxy[1.0.0, 2.0.1, ..., 2.1.1, 3.0.0, ..., 3.3.4, 4.0.0, ..., 4.4.2].

same error

**Comment by tisuchi** [comment]
> > when in laravel 10 i went to use it it shows this error
> > Problem 1 - Root composer.json requires illuminate/contracts 6.0, found illuminate/contracts[v6.0.0] but these were not loaded, likely because it conflicts with another require. Problem 2 - fideloper/proxy 1.0.0 requires illuminate/support 4.0.x -> found illuminate/support[v4.0.0, ..., v4.0.10] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[2.0.1, ..., 2.0.2] require illuminate/support ~4 -> found illuminate/support[v4.0.0, ..., v4.2.17] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[2.0.3, ..., 2.1.1] require illuminate/support ~4.0|~5.0 -> found illuminate/support[v4.0.0, ..., v4.2.17, v5.0.0, ..., v5.8.36] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[3.0.0, ..., 3.3.4, 4.0.0, ..., 4.1.0] require illuminate/contracts ~5.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36] but it conflicts with your root composer.json require (6.0). - fideloper/proxy 4.2.0 requires illuminate/contracts ~5.0|~6.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[4.2.1, ..., 4.2.2] require illuminate/contracts ^5.0|^6.0|^7.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[4.3.0, ..., 4.4.0] require illuminate/contracts ^5.0|^6.0|^7.0|^8.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but these were not loaded, likely because it conflicts with another require. - fideloper/proxy[4.4.1, ..., 4.4.2] require illuminate/contracts ^5.0|^6.0|^7.0|^8.0|^9.0 -> found illuminate/contracts[v5.0.0, ..., v5.8.36, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.7] but these were not loaded, likely because it conflicts with another require. - Root composer.json requires fideloper/proxy * -> satisfiable by fideloper/proxy[1.0.0, 2.0.1, ..., 2.1.1, 3.0.0, ..., 3.3.4, 4.0.0, ..., 4.4.2].
> 
> same error

@summerKK Does it help you?
https://github.com/laravel/framework/issues/40883#issuecomment-1556011494

**Comment by summerKK** [comment]
> 

My description was unclear. 
Today, I encountered a similar error when upgrading from Laravel 6.0 to Laravel 7.0. However, my error message was: 
`- laravel/framework[v8.74.0, ..., v8.83.27] require symfony/http-foundation ^5.4 -> found symfony/http-foundation[v5.4.0, ..., v5.4.24] but these were not loaded, likely because it conflicts with another require.`
 I used `composer why-not symfony/http-foundation "^5.4"` to identify other dependencies. Upgrading those packages resolved the issue.
